### PR TITLE
JSON message parsing

### PR DESCRIPTION
**Pull request by chenziliang**
modified:   drain/logging_splunk.go
	modified:   drain/logging_splunk_test.go

### CODE REVIEW

**Comment by chenziliang** [comment]
screen shot 2017-07-20 at 7 35 28 am

After enhancement, user can see json (message) events in Splunk

**Comment by chenziliang** [comment]
Since it checks before do JSON parsing, I would expect very minor perf impact according to previous AddOn experience

File: drain/logging_splunk.go [hunk 1]
```diff
@@ -86,9 +87,35 @@ func (l *LoggingSplunk) indexEvents(client splunk.SplunkClient, batch []map[stri
 	return nil
 }
 
+// ToJSON tries to detect the JSON pattern for msg first, if msg contains JSON pattern either
+// a map or an array (for efficiency), it will try to convert msg to a JSON object. If the convertion
+// success, a JSON object will be returned. Otherwise the original msg will be returned
+// If the msg param doesn't contain the JSON pattern, the msg will be returned directly
+func ToJson(msg string) interface{} {
+	trimed := strings.Trim(msg, " ")
```
Comment (cholick) [inline drain/logging_splunk.go]:
```
trimed := strings.TrimSpace(msg)
```
 would be a little more robust, trimming all whitespace.

Also, trimed should be spelled trimmed :)

Comment (chenziliang) [inline drain/logging_splunk.go]: strings.Trim(msg, " ") will trim all spaces, unit test already covers this. strings.TrimSpace is far readable. Fixed "trimed"

**Pull request merged**